Automattic/newspack-listings
# perf: enqueue front-end assets only when necessary

### All Submissions:






### Changes proposed in this Pull Request:



Splits front-end assets by feature and only enqueues them when the corresponding feature is present on the current page. Should save a bit of CSS and JS loading on the front-end when not using certain Listings features. This should especially help with AMP CSS limits as it will reduce the amount of CSS loaded on nearly every page by a small amount.

Also fixes a conflict on non-AMP pages between our non-AMP JS and `mediaelement`'s JS.

### How to test the changes in this Pull Request:

This is a pure refactor and shouldn't result in any different functional behavior. To test:

1. Check out this branch and delete the `dist` folder with `rm -r dist`.
2. Run `npm run build` to rebuild assets.
3. Test the following to ensure the front-end assets still work as expected:
  - A post in both AMP and non-AMP mode with a Curated List block using the "Load More" feature
  - An archive page (e.g. category) with only listings, while the "Show listing archives as grid" option is enabled in Listings > Settings 
  - A singular Marketplace listing using one of the the Real Estate block patterns
  - A singular Event listing using the Event Dates block
  - (optional) The Order Details page while logged in as a self-serve customer (must purchase a self-serve listing first—place a self-serve listings block on a page and complete a purchase as a non-logged-in user)

#### Conflict with `mediaelement` JS

1. Add a Jetpack Podcast Player to a post and configure it with a valid URL (try: https://feeds.megaphone.fm/no-place-like-home). Also add a query-based Curated List block with "load more" functionality to the same post.
2. Disable AMP on the post.
3. View the post in an incognito window. On `master`, observe that the podcast player block is missing the media player component.
4. Check out this branch and refresh. Confirm that the media player loads correctly now, and that Curated List still works as expected.

### Other information:
